src/server_mocker/server_mocker_error.rs

```diff
@@ -6,73 +6,64 @@
 //!
 //! If so, errors can be retrieved with [`ServerMocker::pop_server_error`](crate::server_mocker::ServerMocker::pop_server_error) method.
 
-use std::error::Error;
-use std::fmt::{Display, Formatter};
-
-/// Represents the fatalities of a server mocker error.
-#[derive(Debug, PartialEq, Eq, Copy, Clone)]
-pub enum ServerMockerErrorFatality {
-    /// The error is fatal, the server mocker will stop.
-    Fatal,
-    /// The error is not fatal, the server mocker will continue.
-    NonFatal,
-}
-
-impl ServerMockerErrorFatality {
-    /// Returns true if the error is fatal, false otherwise.
-    pub fn is_fatal(&self) -> bool {
-        match self {
-            Self::Fatal => true,
-            Self::NonFatal => false,
-        }
-    }
-}
-
-/// Will display "Fatal" or "Non-fatal" depending on the error fatality.
-impl Display for ServerMockerErrorFatality {
-    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        match self {
-            Self::Fatal => write!(f, "Fatal"),
-            Self::NonFatal => write!(f, "Non fatal"),
-        }
-    }
-}
+use crate::server_mocker_instruction::Instruction;
+use std::io;
+use std::net::SocketAddr;
+use std::sync::mpsc::SendError;
 
 /// Represents an error raised by a server mocker.
 ///
 /// The error is raised directly during call to [`ServerMocker`](crate::server_mocker::ServerMocker) methods, or when the server mocker is running asynchronously and an error occurs.
 ///
 /// If so, errors can be retrieved with [`ServerMocker::pop_server_error`](crate::server_mocker::ServerMocker::pop_server_error) method.
-#[derive(Debug, Clone, PartialEq, Eq)]
-pub struct ServerMockerError {
-    /// The error message.
-    pub message: String,
-    /// The error [fatality](ServerMockerErrorFatality) - fatal if the mocked server stopped.
-    pub fatality: ServerMockerErrorFatality,
+#[derive(Debug, thiserror::Error)]
+#[allow(missing_docs)]
```

Comment:
I think it is worth to add doc even here, cf my previous comment about the warning about missing_docs :smile: